### PR TITLE
Delete character functionality

### PR DESCRIPTION
**Pull request by antgreen13**
Add functionality to delete a character

This change will add a UI interaction that will allow users to delete one of the characters they created. The UI button will make a http request that will delete the character document from firebase.

### CODE REVIEW

File: src/app/home/home.component.html [hunk 1]
```diff
@@ -22,6 +22,7 @@ <h4 class="card-title">Welcome {{ user["name"] }}!</h4>
         Add a character
       </button>
     </div>
+
```
Comment (ericchen94) [inline src/app/home/home.component.html]: nit: remove newline

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -79,4 +81,15 @@ export class HomeComponent implements OnInit, OnDestroy {
...
+
+  public removeCharacter(index: number) {
+    if (this.user) {
+      this.apiService.deleteCharacter(this.user.email, this.dataArray[index].name).subscribe(result => {
+        this.dataArray.splice(index, 1);
+        this.characterData.data =[...this.dataArray]
+      })
+    }
+    
+
```
Comment (ericchen94) [inline src/app/home/home.component.ts]: nit: remove newlines

File: src/app/home/home.component.ts [hunk 3]
```diff
@@ -26,6 +26,8 @@ export class HomeComponent implements OnInit, OnDestroy {
   private subs = new Subscription();
 
   displayedColumns: string[] = ['name', 'race', 'class'];
+  columnsToDisplayWithExpand = [...this.displayedColumns, 'delete'];
```
Comment (ericchen94) [inline src/app/home/home.component.ts]: not needed but naming could be changed to be similar to displayedColumns, maybe displayedExpandedColumns ?

**Review by ericchen94 — CHANGES_REQUESTED** [review]
One suggestion - you can also add a screenshot of the added component(s) so the reviewer can visually see the changes being made

File: src/app/home/home.component.ts (hunk 3 above)
Comment (antgreen13) [reply src/app/home/home.component.ts]: replaced with displayedActionColunms because the additional columns will be edit and delete buttons.

**Review by ericchen94 — APPROVED** [review]

**Pull request merged**